blacklist.go

```diff
@@ -34,8 +34,7 @@ func (b MapBlacklist) Contains(p peer.ID) bool {
 
 // TimeCachedBlacklist is a blacklist implementation using a time cache
 type TimeCachedBlacklist struct {
-	sync.RWMutex
-	tc *timecache.TimeCache
+	tc timecache.TimeCache
```

Comment:
Since the ported time cache implementation is thread-safe, I thought it would be ok to remove the `RWMutex` from here.

Reply:
There is a minor change in the atomicity of operations though. Before, the lock was held across the entire `TimeCachedBlacklist.Add()` call, and thus across both the `tc.Has()` and `tc.Add()` calls.

Now, the time cache lock will be held only across the individual `tc.Has()` and `tc.Add()` calls. I think that should be ok, but I can totally just revert this change.

Reply:
One more thing - I retained the original (first-seen) time cache implementation here, but could also switch it to the new one.

Reply:
> One more thing - I retained the original (first-seen) time cache implementation here, but could also switch it to the new one.

Ok, scratch this. As per @lidel's comment, I aligned everything to use the new implementation.